streams/src/test/java/org/apache/kafka/streams/integration/RackAwarenessIntegrationTest.java

```diff
@@ -0,0 +1,340 @@
+            kafkaStreamsWithConfiguration.kafkaStreams.close(Duration.ofMillis(IntegrationTestUtils.DEFAULT_TIMEOUT));
+            IntegrationTestUtils.purgeLocalStreamsState(kafkaStreamsWithConfiguration.configuration);
+        }
+        kafkaStreamsInstances.clear();
+    }
+
+    @Test
+    public void shouldDistributeStandbyReplicasBasedOnClientTags() throws Exception {
+        final Topology topology = createStatefulTopology();
+        final int numberOfStandbyReplicas = 1;
+
+        createAndStart(topology, buildClientTags("eu-central-1a", "k8s-cluster-1"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+        createAndStart(topology, buildClientTags("eu-central-1a", "k8s-cluster-1"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+        createAndStart(topology, buildClientTags("eu-central-1a", "k8s-cluster-1"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+
+        createAndStart(topology, buildClientTags("eu-central-1b", "k8s-cluster-1"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+        createAndStart(topology, buildClientTags("eu-central-1b", "k8s-cluster-1"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+        createAndStart(topology, buildClientTags("eu-central-1b", "k8s-cluster-1"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+
+        waitUntilRackAwareTaskDistributionIsReached(TAG_ZONE);
+    }
+
+    @Test
+    public void shouldDistributeStandbyReplicasOverMultipleClientTags() throws Exception {
+        final Topology topology = createStatefulTopology();
+        final int numberOfStandbyReplicas = 2;
+
+        createAndStart(topology, buildClientTags("eu-central-1a", "k8s-cluster-1"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+        createAndStart(topology, buildClientTags("eu-central-1b", "k8s-cluster-1"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+        createAndStart(topology, buildClientTags("eu-central-1c", "k8s-cluster-1"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+
+        createAndStart(topology, buildClientTags("eu-central-1a", "k8s-cluster-2"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+        createAndStart(topology, buildClientTags("eu-central-1b", "k8s-cluster-2"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+        createAndStart(topology, buildClientTags("eu-central-1c", "k8s-cluster-2"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+
+        createAndStart(topology, buildClientTags("eu-central-1a", "k8s-cluster-3"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+        createAndStart(topology, buildClientTags("eu-central-1b", "k8s-cluster-3"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+        createAndStart(topology, buildClientTags("eu-central-1c", "k8s-cluster-3"), asList(TAG_ZONE, TAG_CLUSTER), numberOfStandbyReplicas);
+
```

Comment:
I think, here it makes sense to wait until all Streams clients are `RUNNING` so that we know that the rebalance is done.